java/compilers/CFGReader.java

```diff
@@ -19,11 +19,11 @@ private static CFG readGrammar(File cfg) {
 
             while (in.hasNext()) {
                 String line = in.nextLine();
-                if(line.trim().isBlank()) {
+                if(line.isBlank()) {
```

Comment:
I did not see this behavior, unless it is some sort of edge case. If a line contains only whitespace, trimming the leading and trailing whitespace should not affect anything.

Reply:
I was thinking that it would allow a whitespace-only line into this block, and then line 34 would throw.  It looks like `String#isBlank` checks if the line is only whitespace so there isn't an issue.